mne/io/eyelink/_utils.py

```diff
@@ -5,6 +5,252 @@
 import re
 import numpy as np
 
+from ...utils import _check_pandas_installed
+
+
+def _isfloat(token):
+    """Boolean test for whether string can be of type float.
+
+    Parameters
+    ----------
+    token : str
+        Single element from tokens list.
+    """
+    if isinstance(token, str):
+        try:
+            float(token)
+            return True
+        except ValueError:
+            return False
+    else:
+        raise ValueError(
+            "input should be a string," f" but {token} is of type {type(token)}"
+        )
+
+
+def _convert_types(tokens):
+    """Convert the type of each token in list.
+
+    The tokens input is a list of string elements.
+    Posix timestamp strings can be integers, eye gaze position and
+    pupil size can be floats. flags token ("...") remains as string.
+    Missing eye/head-target data (indicated by '.' or 'MISSING_DATA')
+    are replaced by np.nan.
+
+    Parameters
+    ----------
+    Tokens : list
+        List of string elements.
+
+    Returns
+    -------
+        Tokens list with elements of various types.
+    """
+    return [
+        int(token)
+        if token.isdigit()  # execute this before _isfloat()
+        else float(token)
+        if _isfloat(token)
+        else np.nan
+        if token in (".", "MISSING_DATA")
+        else token  # remains as string
+        for token in tokens
+    ]
+
+
+def _parse_line(line):
+    """Parse tab delminited string from eyelink ASCII file.
+
+    Takes a tab deliminited string from eyelink file,
+    splits it into a list of tokens, and converts the type
+    for each token in the list.
+    """
+    if len(line):
+        tokens = line.split()
+        return _convert_types(tokens)
+    else:
+        raise ValueError("line is empty, nothing to parse")
```

Comment:
For stuff like this I think the short-circuit is easier to read (and involves less indentation, which is always nice)
```suggestion
    if not len(line):
        raise ValueError("line is empty, nothing to parse")
    tokens = line.split()
    return _convert_types(tokens)
```
But also we should try to keep coverage up (95%+ if possible, at 90 currently) and the valueerror branch is never hit:

https://app.codecov.io/gh/mne-tools/mne-python/pull/11746/blob/mne/io/eyelink/_utils.py#L72

Can we avoid the check altogether? It seems like `_convert_types` would just return an empty list in the case of `tokens` being `[]` which seems okay

Reply:
Yeah, I agree we can always assume that the line will never be empty (I put that in there during development as an abundance of caution).

Reply:
I've been wondering if it would be a good idea to take another pass through `read_raw_eyelink` during my GSoC and see where we can clean things up. 

One thing is we're already requiring pandas for the function so maybe we could leverage vectorization more. 

I'm not sure how close of a look you got to take during the initial PR review. What do you think? worth it?